airstack/token.go

```diff
@@ -0,0 +1,84 @@
+package airstack
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	gql "github.com/vocdoni/vote-frame/airstack/graphql"
+)
+
+// TokenDetails wraps useful information about a token
+type TokenDetails struct {
+	Decimals    int8
+	Name        string
+	Symbol      string
+	TotalSupply *big.Int
+}
+
+// GetTokenDetails gets a token details given its address and blockchain
+func (c *Client) GetTokenDetails(
+	tokenAddress common.Address,
+	blockchain gql.TokenBlockchain,
+) (*TokenDetails, error) {
+	cctx, cancel := context.WithTimeout(c.ctx, apiTimeout)
+	defer cancel()
+	resp, err := gql.GetTokenDetails(cctx, c.Client, tokenAddress, blockchain)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get token details from Airstack: %w", err)
+	}
+	totalSupply := new(big.Int)
+	totalSupply.SetString(resp.Token.TotalSupply, 10)
+	return &TokenDetails{
+		Decimals:    int8(resp.Token.Decimals),
+		Name:        resp.Token.Name,
+		Symbol:      resp.Token.Symbol,
+		TotalSupply: totalSupply,
+	}, nil
+}
+
+// TokenHolder wraps a token holder with its address and balance of a certain token
+type TokenHolder struct {
+	Address common.Address
+	Balance *big.Int
+}
+
+// getTokenBalances is a wrapper around the generated function for GraphQL query GetTokenBalances.
+func (c *Client) getTokenBalances(
+	tokenAddress common.Address,
+	blockchain gql.TokenBlockchain,
+	limit int,
+	cursor string,
+) (*gql.GetTokenBalancesResponse, error) {
+	cctx, cancel := context.WithTimeout(c.ctx, apiTimeout)
+	defer cancel()
+	return gql.GetTokenBalances(cctx, c.Client, tokenAddress, blockchain, limit, cursor)
+}
+
+// GetTokenBalances gets all the token holders of a given token in a given chain
+// calling the Airstack API. This function also takes care of Airstack API pagination.
+func (c *Client) GetTokenBalances(tokenAddress common.Address, blockchain gql.TokenBlockchain) ([]*TokenHolder, error) {
```

Comment:
How is `gql.TokenBlockchain` obtained? This is a bit weird because on one side the Client abstracts the consumer from the GraphQL layer (i.e. returning custom types such as TokenHolder), but then it requires a specific graphQL parameter. So I'd either abstract everything (the caller does not know about graphQL) or the opposite.

Reply:
This was made because as the graphql schema is being updated time to time, the supported networks can change too. The solution to this is to create constants equal to the ones provided to the graphql schema (abstract everything). 
The problem I see with this approach is that is very easy to forget adding the new constants if the schema changed. The way it is currently done will always have the autogenerated code, so the TokenBlockchain will always be up to date.

But given that this is not a neat approach, I am going to abstract everything and add a note on the README to keep an eye on networks available.